.github/workflows/infrastructure.yml

```diff
@@ -23,150 +23,418 @@
 #
 # TO-DO
 # *****
-# 0. Add --internal-only true \ to CAE
 # 1. Add delete locks
 # 2. Add log workspace
 # 3. Add Insight
 # 4. Add monitor
-# 5. Add API management service
-# 6. Add Key vault
-# 7. Add VNET peer (x-subscription RBAC SP)
+# 5. Add Key vault
 
 name: Infrastructure 🔨
 run-name: APIM base infrastructure build from ${{ github.repository }}
 
 on:
   push:
-    branches: [ infrastructure ]
+    branches:
+      - infrastructure
     paths: [ .github/workflows/infrastructure.yml ]
 
 env:
   product: apim
   environment: infrastructure
   timezone: 'Europe/London'
-  # Deployment environment target i.e., `dev`, `staging`, `prod` 
+  # Deployment environment target i.e., `dev`, `staging`, `prod`
   target: ${{ vars.environment }}
 
 jobs:
-
-# 1. Setup infrastructure variables
+  # 1. Setup infrastructure variables
   setup:
     name: Setup 🔧
     runs-on: [ self-hosted, APIM, infrastructure ]
     outputs:
       environment: ${{ env.environment }}
       timezone: ${{ env.timezone }}
     steps:
-    - name: Environment 🧪
-      run: echo "Environment set to ${{ env.environment }}"
+      - name: Environment 🧪
+        run: echo "Environment set to ${{ env.environment }}"
 
-    - name: Timezone 🌐
-      run: echo "Timezone set to ${{ env.timezone }}"
+      - name: Timezone 🌐
+        run: echo "Timezone set to ${{ env.timezone }}"
 
-# 2. Base infrastructure creation
+  # 2. Base infrastructure creation
   base:
     name: Base 🧱
     needs: setup
     environment: ${{ needs.setup.outputs.environment }}
+    outputs:
+      environment: ${{ env.environment }}
     runs-on: [ self-hosted, APIM, infrastructure ]
     steps:
+      - name: Pre-production 💫
+        if: contains('["dev", "staging"]', env.target)
+        run: echo "type=Preproduction" >> $GITHUB_ENV
+
+      - name: Production 💫
+        if: ${{ 'prod' == env.target }}
+        run: echo "type=Production" >> $GITHUB_ENV
+
+      - name: Tags 🏷️
+        run: echo tags='Environment=${{ env.type }}' \
+          'Product=${{ env.product }}' \
+          'Team=development' >> $GITHUB_ENV
+
+      - name: Login 🔐
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Azure defaults ✨
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            # Basic
+            az configure --defaults location=${{ vars.REGION }}
+            az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
```

Comment:
Can we use `001` from a variable? Something like... `version`. This because we may need to create new variations for it in the futur. I.e. `002`

Reply:
Good shout, will do.

Reply:
Have introduced `var.VERSION` variable, this can also be further used for iterative infrastructure.